dspace-server-webapp/src/main/java/org/dspace/app/rest/repository/ClarinUserMetadataRestController.java

```diff
@@ -303,19 +308,87 @@ private void sendEmailWithDownloadLink(Context context, DSpaceObject dso,
-            log.error("Cannot send the email because: " + e.getMessage());
+            log.error("Cannot send the email with download link, because: " + e.getMessage());
+            throw new MessagingException(e.getMessage());
+        }
+        // If previous mail fails with exception, this block never executes = admin is NOT
+        // notified, if the mail is not really sent (if it fails HERE, not later, e.g. due to mail server issue).
+        sendAdminNotificationEmail(context, downloadLink, dso, clarinLicense, mailType, clarinUserMetadataRestList);
+
+    }
+
+    private void sendAdminNotificationEmail(Context context,
+                                            String downloadLink,
+                                            DSpaceObject dso,
+                                            ClarinLicense clarinLicense,
+                                            MailType mailType,
+                                            List<ClarinUserMetadataRest> extraMetadata)
+            throws MessagingException, IOException {
+        try {
+            Locale locale = context.getCurrentLocale();
+            Email email2Admin = Email.getEmail(I18nUtil.getEmailFilename(locale, "clarin_download_link_admin"));
+            String ccAdmin = configurationService.getProperty("download.email.cc");
+            List<String> ccEmails = new ArrayList<>();
+            if (ccAdmin != null && !ccAdmin.isEmpty()) {
+                ccEmails.add(ccAdmin);
+            }
+            String licenseName = clarinLicense.getName().trim().replace(" ", "_").toLowerCase();
+            String[] licenseSpecialRecipients = configurationService.getArrayProperty("download.email.cc."
+                    + licenseName);
+            if (licenseSpecialRecipients != null) {
+                for (String cc : licenseSpecialRecipients) {
+                    if (!cc.isEmpty()) {
+                        ccEmails.add(cc.trim());
+                    }
+                }
+            }
+
+            if (!ccEmails.isEmpty()) {
+                for (String cc : ccEmails) {
+                    email2Admin.addRecipient(cc);
+                }
+                if (mailType == MailType.ALLZIP) {
+                    email2Admin.addArgument("all files requested");
+                } else if (mailType == MailType.BITSTREAM) {
+                    email2Admin.addArgument(dso.getName());
+                } else {
+                    throw new IllegalArgumentException("The mail type is not supported.");
+                }
+                email2Admin.addArgument(downloadLink);
+                email2Admin.addArgument(clarinLicense.getDefinition());
+                if (context.getCurrentUser() != null) {
+                    email2Admin.addArgument(context.getCurrentUser().getFullName());
+                    email2Admin.addArgument(context.getCurrentUser().getEmail());
+
+                } else {
+                    email2Admin.addArgument("Anonymous user");
```

Comment:
2x ?

Reply:
Yes, because if it is not anonymous user, we add name and then email. Both are replaced with string "Anonymous user" when it is an anonymous download. They have to be separate, because in the other case, they have different values. Therefore here, it has to be 2 times. Even if it is done differently, it will still be two times. Added comments.